src/core/src/consensus_state_store.rs

```diff
@@ -67,6 +68,129 @@ impl<'a> ConsensusStateStore<'a> {
+mod tests {
+    use super::*;
+    use database::config;
+
+    /// Asserts that there are COUNT many objects in DB.
+    fn assert_database_db_count(count: usize, db: &LmdbDatabase) {
+        let reader = db.reader().unwrap();
+
+        assert_eq!(reader.count().unwrap(), count,);
+    }
+
+    /// Asserts that there are are COUNT many objects in DB's INDEX.
+    fn assert_index_count(index: &str, count: usize, db: &LmdbDatabase) {
+        let reader = db.reader().unwrap();
+
+        assert_eq!(reader.index_count(index).unwrap(), count,);
+    }
+
+    /// Asserts that KEY is associated with VAL in DB.
+    fn assert_key_value(key: u8, val: u8, db: &LmdbDatabase) {
+        let reader = db.reader().unwrap();
+
+        assert_eq!(reader.get(&[key]).unwrap(), [val],);
+    }
+
+    /// Asserts that KEY is associated with VAL in DB's INDEX.
+    fn assert_index_key_value(index: &str, key: u8, val: u8, db: &LmdbDatabase) {
+        let reader = db.reader().unwrap();
+
+        assert_eq!(reader.index_get(index, &[key]).unwrap().unwrap(), [val],);
+    }
+
+    /// Asserts that KEY is not in DB.
+    fn assert_not_in_database_db(key: u8, db: &LmdbDatabase) {
+        let reader = db.reader().unwrap();
+
+        assert!(reader.get(&[key]).is_none());
+    }
+
+    /// Asserts that KEY is not in DB's INDEX.
+    fn assert_not_in_index(index: &str, key: u8, db: &LmdbDatabase) {
+        let reader = db.reader().unwrap();
+
+        assert!(reader.index_get(index, &[key]).unwrap().is_none());
+    }
+
+    fn create_context() -> Result<lmdb::LmdbContext, CliError> {
+        let path_config = config::get_path_config();
+        let statestore_path = &path_config.data_dir.join(config::get_filename());
+
+        lmdb::LmdbContext::new(statestore_path, 1, None)
+            .map_err(|err| CliError::EnvironmentError(format!("{}", err)))
+    }
+
+    #[test]
+    fn test_state_store_get() {
+
+        let mut ctx = create_context().unwrap();
+        let statestore_db = lmdb::LmdbDatabase::new(
+            &ctx,
+            &["index_consensus_state"],
+        ).map_err(|err| CliError::EnvironmentError(format!("{}", err))).unwrap();
+
+        let mut state_store = ConsensusStateStore::new(statestore_db);
+
+        assert!(state_store.get(BlockId::from(vec![11])).is_err());
+        state_store.put( BlockId::from(vec![11]), ConsensusState::default() );
+
+        assert!(state_store.get(BlockId::from(vec![11])).is_ok());
+        //cleanup
+        state_store.delete( BlockId::from(vec![11]) );
+    }
+
+    #[test]
+    fn test_state_store_put() {
+
+        let mut ctx = create_context().unwrap();
+        let statestore_db = lmdb::LmdbDatabase::new(
+            &ctx,
+            &["index_consensus_state"],
+        ).map_err(|err| CliError::EnvironmentError(format!("{}", err))).unwrap();
+
+        let mut state_store = ConsensusStateStore::new(statestore_db);
+
+        assert!(state_store.get(BlockId::from(vec![13])).is_err());
```

Comment:
What does index 13 correspond to. can we avoid hard codings or add comments

Reply:
These are all unit tests. Hence hard coded values are seen too often. These hard coded values are dummy/mocked variables only which do not impact the test result.